06.bowling_object/frame.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require_relative 'shot'
+
+class Frame
+  attr_reader :shots
+
+  def initialize(frame_number, first_shot, second_shot = nil, third_shot = nil)
+    @frame_number = frame_number
+    @shots = [first_shot]
+    @shots << second_shot unless second_shot.nil?
+    @shots << third_shot unless third_shot.nil?
+  end
+
+  def calc_score(frames)
+    result = shots.sum(&:score)
+    if @frame_number < 9
+      if strike?
+        result += strike_score_bonus(frames)
+      elsif spare?
+        result += spare_score_bonus(frames)
+      end
+    end
+    result
+  end
+
+  protected
+
+  def strike?
+    shots[0].score == 10
+  end
+
+  def spare?
+    !strike? && shots.first(2).sum(&:score) == 10
+  end
+
+  def strike_score_bonus(frames)
```

Comment:
```suggestion
  def calc_strike_bonus(frames)
```

メソッドは動詞で始める方がいいですね

https://qiita.com/jnchito/items/459d58ba652bf4763820